### PR TITLE
Added ARKitFullExample with both SpriteKit and SceneKit

### PR DESCRIPTION
**Pull request by AFathi**
In this PR, I have added an ARKit example project titled `ARKitFullExample` that include both AR SpriteKit and SceneKit rendering.

This example project is using [ARVideoKit](https://github.com/AFathi/ARVideoKit) to render both AR SpriteKit and SceneKit. I built ARVideoKit to take away the complex process needed to retrieve fully rendered ARKit buffers. 

I have also updated the `Podfile` to simply use `pod install` and add the project to `VideoQuickStart.xcworkspace`.

If you're interested to know more about ARVideoKit, you may check out the [framework's documentation](https://github.com/AFathi/ARVideoKit/wiki) or email [me](mailto:me@ahmedbekhit.com)@ahmedbekhit.com

### CODE REVIEW

**Comment by ptankTwilio** [comment]
@afathi We liked your example of TwilioVideo usage with SpriteKit and SceneKit. This is a great contribution. Thank you for submitting the pull request. We have the following feedback - 

1. While running the app in a bidirectional audio+video call on iPhone 6S plus, the CPU utilization was very high, >140 % approximately (see the snapshot below). Given that our existing [ARKitExample](https://github.com/twilio/video-quickstart-swift/tree/master/ARKitExample) app takes around 90% of CPU, we are suspecting the performance issue is caused by [ARVideoKit](https://github.com/AFathi/ARVideoKit)  framework. Can you take a look and fix the performance issue? 
screen shot 2017-11-22 at 11 44 16 am

2. Per our policy, if a third party framework is not an open source framework, our sample app(s) cannot consume it. However, if you create a project for this app on your repository or a separate repository, we would love to shout out and post a link to it on our sample apps [README](https://github.com/twilio/video-quickstart-swift/blob/master/README.md). Also we would love to help review the code.

Please let me know if you have any questions. 

Thank you!

File: ARKitFullExample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+		FB0BECB81FBFD45900897207 /* Scene.sks in Resources */ = {isa = PBXBuildFile; fileRef = FB0BECB71FBFD45900897207 /* Scene.sks */; };
+		FB0BECBA1FBFD45900897207 /* Scene.swift in Sources */ = {isa = PBXBuildFile; fileRef = FB0BECB91FBFD45900897207 /* Scene.swift */; };
+		FB0BECBF1FBFD45900897207 /* Main.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = FB0BECBD1FBFD45900897207 /* Main.storyboard */; };
+		FB0BECC11FBFD45900897207 /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = FB0BECC01FBFD45900897207 /* Assets.xcassets */; };
+		FB0BECC41FBFD45900897207 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = FB0BECC21FBFD45900897207 /* LaunchScreen.storyboard */; };
+		FB48F3C51FC125AC005D4965 /* ARVideoKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = FB48F3C11FC12583005D4965 /* ARVideoKit.framework */; };
+		FB48F3C61FC125AC005D4965 /* ARVideoKit.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = FB48F3C11FC12583005D4965 /* ARVideoKit.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Comment (ptankTwilio) [inline ARKitFullExample.xcodeproj/project.pbxproj]: Generally checking in frameworks is not a good idea.

File: ARKitFullExample/View Controllers/SCNViewController.swift [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    }
+}
+
+//MARK: - ARVideoKit Renderer Method
+extension SCNViewController: RenderARDelegate {
+    func frame(didRender buffer: CVPixelBuffer, with time: CMTime, using rawBuffer: CVPixelBuffer) {
+        if isStreamingVideo {
+            self.frame = TVIVideoFrame(timestamp: Int64(CMTimeGetSeconds(time)),
+                                       buffer: buffer,
+                                       orientation: TVIVideoOrientation.up)
```
Comment (ptankTwilio) [inline ARKitFullExample/View Controllers/SCNViewController.swift]: `TVIVideoFrame` takes timestamp in nano-seconds - [example usage](https://github.com/twilio/video-quickstart-swift/blob/master/ARKitExample/ViewController.swift#L91-L93)

File: ARKitFullExample/View Controllers/SKViewController.swift [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    }
+}
+
+//MARK: - ARVideoKit Renderer Method
+extension SKViewController: RenderARDelegate {
+    func frame(didRender buffer: CVPixelBuffer, with time: CMTime, using rawBuffer: CVPixelBuffer) {
+        if isStreamingVideo {
+            self.frame = TVIVideoFrame(timestamp: Int64(CMTimeGetSeconds(time)),
+                                       buffer: buffer,
+                                       orientation: TVIVideoOrientation.up)
```
Comment (ptankTwilio) [inline ARKitFullExample/View Controllers/SKViewController.swift]: `TVIVideoFrame` takes timestamp in nano-seconds - [example usage](https://github.com/twilio/video-quickstart-swift/blob/master/ARKitExample/ViewController.swift#L91-L93)

**Comment by AFathi** [comment]
@ptankTwilio,
Thank you for the great feedback! 

Would the timestamp be one of the factors that slows down the performance? Or would it be mainly caused from the way I'm rendering the scenes in my framework (ARVideoKit)?

-Ahmed

**Comment by ptankTwilio** [comment]
@AFathi I was getting high CPU utilization with the fixed timestamp also. I am assuming the high cpu utilization is caused by ARVideoKit rendering. I would encourage you to monitor the CPU utilization and compare with our existing ARKitExample performance. If the ARVideoKit framework is getting the pixel buffers without taking an ARSCNView's snapshot, you should see a better performance than our ARKitExample. You can try running a profiler to see what exactly is causing the higher CPU usage. 
 
I am excited to try out again and see the performance results once you have this fixed.

Let me know if you have any questions.

Best,
-Piyush

**Comment by ceaglest** [comment]
Closing this PR due to inactivity, and major changes on master since it was posted.

**Pull request closed without merging**